spring-data-relational/src/main/java/org/springframework/data/relational/core/mapping/AggregatePath.java

```diff
@@ -300,27 +304,114 @@ static TableInfo of(AggregatePath path) {
+				ColumInfosBuilder ciBuilder = new ColumInfosBuilder(idPath);
+				idPath.getRequiredLeafEntity().doWithProperties((PropertyHandler<RelationalPersistentProperty>) p -> {
+					AggregatePath idElementPath = idPath.append(p);
+					ciBuilder.add(idElementPath, ColumnInfo.of(idElementPath));
+				});
+				return ciBuilder.build();
+			} else {
+				ColumInfosBuilder ciBuilder = new ColumInfosBuilder(idPath.getParentPath());
+				ciBuilder.add(idPath, ColumnInfo.of(idPath));
+				return ciBuilder.build();
+			}
 		}
-	}
 
-	record ColumnInfo(
+		private static ColumnInfos computeReverseColumnInfo(AggregatePath path) {
+
+			AggregatePath tableOwner = AggregatePathTraversal.getTableOwningPath(path);
+
+			if (tableOwner.isRoot()) {
+				return ColumnInfos.empty(tableOwner);
+			}
+
+			AggregatePath idDefiningParentPath = tableOwner.getIdDefiningParentPath();
+			RelationalPersistentProperty leafProperty = tableOwner.getRequiredLeafProperty();
+
+			RelationalPersistentProperty idProperty = idDefiningParentPath.getLeafEntity().getIdProperty();
+
+			if (idProperty != null) {
+				if (idProperty.isEntity()) {
+
+					AggregatePath idBasePath = idDefiningParentPath.append(idProperty);
+					ColumInfosBuilder ciBuilder = new ColumInfosBuilder(idBasePath);
+
+					RelationalPersistentEntity<?> idEntity = idBasePath.getRequiredLeafEntity();
+					idEntity.doWithProperties((PropertyHandler<RelationalPersistentProperty>) p -> {
+						AggregatePath idElementPath = idBasePath.append(p);
+						SqlIdentifier name = idElementPath.getColumnInfo().name();
+						name = name.transform(n -> idDefiningParentPath.getTableInfo().qualifiedTableName.getReference() + "_" + n);
+
+						ciBuilder.add(idElementPath, name, name);
+					});
+
+					return ciBuilder.build();
+
+				} else {
+
+					ColumInfosBuilder ciBuilder = new ColumInfosBuilder(idDefiningParentPath);
+					SqlIdentifier reverseColumnName = leafProperty
+							.getReverseColumnName(idDefiningParentPath.getRequiredLeafEntity());
+
+					ciBuilder.add(idProperty, reverseColumnName,
+							AggregatePathTableUtils.prefixWithTableAlias(path, reverseColumnName));
+
+					return ciBuilder.build();
+				}
+			} else {
 
-			/* The name of the column used to represent this property in the database. */
-			SqlIdentifier name, /* The alias for the column used to represent this property in the database. */
-			SqlIdentifier alias) {
+				ColumInfosBuilder ciBuilder = new ColumInfosBuilder(idDefiningParentPath);
+				SqlIdentifier reverseColumnName = leafProperty
+						.getReverseColumnName(idDefiningParentPath.getRequiredLeafEntity());
+
+				ciBuilder.add(idDefiningParentPath, reverseColumnName,
+						AggregatePathTableUtils.prefixWithTableAlias(path, reverseColumnName));
+
+				return ciBuilder.build();
+			}
+
+		}
+
+		@Deprecated(forRemoval = true)
+		public ColumnInfo reverseColumnInfo() {
+			return reverseColumnInfos.unique();
+		}
+
+		public ColumnInfos effectiveIdColumnInfos() {
```

Comment:
What is effective and why do we deprecate the other method?

Reply:
Clarified in comments